app/lib/features/chat_ng/widgets/events/text_message_event.dart

```diff
@@ -1,153 +1,112 @@
+  Widget build(BuildContext context) {
+    final inner =
+        (isOnlyEmojis(body))
+            ? _buildEmojiMessage(context)
+            : _buildTextMessage(context);
 
-    // Handle emoji messages
-    if (_type == TextMessageType.emoji) {
-      final emojiTextStyle =
-          _isMe
-              ? chatTheme.sentEmojiMessageTextStyle
-              : chatTheme.receivedEmojiMessageTextStyle;
-      return Padding(
-        padding: const EdgeInsets.symmetric(horizontal: 8.0),
-        child: Html(
-          data: body,
-          defaultTextStyle: emojiTextStyle.copyWith(fontFamily: emojiFont),
-        ),
-      );
+    if (widget.isReply) {
+      // we return the widget without any additional reply data
+      return inner;
     }
-    final replied = repliedTo;
 
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: [
-        if (replied != null) ...[replied, const SizedBox(height: 10)],
+        if (widget.repliedTo != null) ...[
+          widget.repliedTo!,
+          const SizedBox(height: 10),
+        ],
+        inner,
+      ],
+    );
+  }
+
+  // emoji only special message
+  Widget _buildEmojiMessage(BuildContext context) => Padding(
+    padding: const EdgeInsets.symmetric(horizontal: 8.0),
+    child: Text(
+      body, // what about formatted body?
+      style: EmojiConfig.emojiTextStyle?.copyWith(fontSize: 28),
+      maxLines: widget.isReply ? 2 : null,
+      overflow: TextOverflow.ellipsis,
+    ),
+  );
+
+  Widget _buildTextMessage(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+    final colorScheme = Theme.of(context).colorScheme;
 
-        Html(
-          linkStyle: TextStyle(
-            color: colorScheme.onPrimary,
-            decoration: TextDecoration.underline,
-          ),
-          shrinkToFit: true,
-          pillBuilder:
-              ({
-                required String identifier,
-                required String url,
-                OnPillTap? onTap,
-              }) => ActerPillBuilder(
-                identifier: identifier,
-                uri: url,
-                roomId: roomId,
-              ),
-          maxLines: _type == TextMessageType.reply ? 2 : null,
-          onLinkTap: (Uri uri) {
-            openUri(ref: ref, uri: uri, lang: L10n.of(context));
-          },
-          defaultTextStyle: textTheme.bodySmall?.copyWith(
-            color:
-                _type == TextMessageType.notice
-                    ? colorScheme.onSurface.withValues(alpha: 0.5)
-                    : colorScheme.onSurface.withValues(alpha: 0.9),
-            overflow:
-                _type == TextMessageType.reply ? TextOverflow.ellipsis : null,
-          ),
-          data: body,
+    final color =
+        widget.isNotice
+            ? colorScheme.onSurface.withValues(alpha: 0.5)
+            : colorScheme.onSurface.withValues(alpha: 0.9);
+
+    final html = bodyFormatted;
+    if (html != null) {
```

Comment:
reply and notice aren't mutually exclusive, but flags to be taken into consideration: a notice can be a reply or have a reply or be emoji's only or be a reply to an emoji only. all this is now possible (before, it wasn't).